osu.Game/Overlays/Direct/PlayButton.cs

```diff
@@ -78,19 +76,42 @@ public PlayButton(BeatmapSetInfo setInfo = null)
                 loadingAnimation = new LoadingAnimation(),
             });
 
-            Playing.ValueChanged += updatePreviewTrack;
+            Playing.ValueChanged += playingStateChanged;
         }
 
         [BackgroundDependencyLoader]
-        private void load(OsuColour colour, AudioManager audio)
+        private void load(OsuColour colour, PreviewTrackManager previewTrackManager)
         {
             hoverColour = colour.Yellow;
-            this.audio = audio;
+            this.previewTrackManager = previewTrackManager;
         }
 
         protected override bool OnClick(InputState state)
         {
-            Playing.Value = !Playing.Value;
+            if (!Playing.Value)
+            {
+                if (Preview == null)
+                {
+                    Task.Run(() =>
```

Comment:
This is worse than the old logic. why is `PreviewTrack` not a component which can be `LoadComponentAsync`'d?